### PR TITLE
Added Export to CSV

### PR DESCRIPTION
**Pull request by ch8n**
# Feature Preview
https://user-images.githubusercontent.com/11576342/107530361-cd3be400-6be1-11eb-9d02-357259f06d4f.mp4

# Current functionality  : 
* Export your transaction details to CSV file 
* This file is located at **Documents/expenso** folder of your device

# How test?
* manually test with sample data

# Limitations?
* This feature isn't supported for android 10+ cause that requires `scoped storage implementation`
* Always open for improvements in structure and solutioning

### CODE REVIEW

File: app/src/main/res/menu/menu_ui.xml [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
+        android:id="@+id/action_about"
+        android:enabled="true"
+        android:title="@string/text_about"
+        app:showAsAction="never" />
+
+    <item
+        android:id="@+id/action_export"
+        android:enabled="true"
+        android:title="@string/text_export_csv"
+        app:showAsAction="never" />
```
Comment (Spikeysanju) [inline app/src/main/res/menu/menu_ui.xml:28]: @ch8n Can you pls move this item below `action_night_mode`

File: app/src/main/java/dev/spikeysanju/expensetracker/services/csv/ExportService.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    val prefix: String = "expenso",
+    val suffix: String = DateFormat
+        .getDateTimeInstance()
+        .format(System.currentTimeMillis())
+        .toString()
+        .replace(" ", "_"),
+    val fileName: String = "$prefix-$suffix.csv",
+    val hostPath: String = Environment
+        .getExternalStorageDirectory()?.absolutePath?.plus("/Documents/Expenso") ?: ""
+)
```
Comment (Spikeysanju) [inline app/src/main/java/dev/spikeysanju/expensetracker/services/csv/ExportService.kt:35]: `.getExternalStorageDirectory()` - It's deprecated. Shall we improve on this? @ch8n

**Review by Spikeysanju — CHANGES_REQUESTED** [review]

File: app/src/main/java/dev/spikeysanju/expensetracker/services/csv/ExportService.kt (hunk 2 above)
Comment (ch8n) [reply app/src/main/java/dev/spikeysanju/expensetracker/services/csv/ExportService.kt:35]: Hi it's deprecated for API level 30 which needs scoped storage I have open an issue for that and currently, the app is blocked to work on this feature on API level 29+

File: app/build.gradle [hunk 3]
```diff
@@ -1,99 +1,102 @@
...
+    kapt "com.google.dagger:hilt-android-compiler:2.30.1-alpha"
+    annotationProcessor 'androidx.hilt:hilt-compiler:1.0.0-alpha03'
+
+    //implementation "com.google.dagger:hilt-android-testing:$hilt_ver"
+    implementation "androidx.hilt:hilt-common:1.0.0-alpha03"
+    implementation "androidx.hilt:hilt-lifecycle-viewmodel:1.0.0-alpha03"
+
+    // OpenCsv
+    implementation "com.opencsv:opencsv:5.3"
+}
```
Comment (Spikeysanju) [inline app/build.gradle:102]: @ch8n Open CSV is causing conflict. Can you pls work on it buddy!

**Review by Spikeysanju — CHANGES_REQUESTED** [review]

**Comment by Spikeysanju** [comment]
@ch8n  We need to show `open` in snackbar to open the CSV file path

File: app/build.gradle (hunk 3 above)
Comment (ch8n) [reply app/build.gradle:102]: common logs were conflicting with android ones fixed it

**Comment by Spikeysanju** [comment]
@ch8n Closing this issue for now. Feel free to make a PR when you fix the above issues. Thanks buddy 🙌.. Have a great day!

**Pull request closed without merging**